### DISCUSSION TITLE
32GB RAM - Filterlist size and DNS Cache size relevant?

### DISCUSSION THREAD

**Discussion by Utini2000** [DNSCrypt/dnscrypt-proxy#2564] [Q&A]
I am running DNSCrypt with filterlist and DNS cache enabled on a laptop with 32GB RAM.
I never reach anywhere close to more than 50% of RAM usage and in the worst case would also have ZRAM SWAP enabled.

I therefor wonder if big filterlists (e.g. oisd big, HaGeZi,...) would cause any slow down on my device?
Also, should I increase DNS cache size since I got enough RAM?

Thanks!

**Comment by jedisct1** [comment]
Lists tend have a lot of redundancy. 

The `generate-domains-blocklist` script will optimize them by removing redundancies:

https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Combining-Blocklists

32 GB RAM is a ton. You will never ever fill that, even with ginormous lists.

**Comment by Utini2000** [comment]
@jedisct1 Thanks! And I guess CPU usage or any other system usage is neg liable on modern hardware (e.g. in my case R7-6850U)? Out of curiosity: Does the filterlist stay in RAM all the time so accessing & sending each request through it will be faster?
